### DISCUSSION TITLE
Solving environment: failed during installation

### DISCUSSION THREAD

**Discussion by DavoodZivar** [idaholab/moose#27665] [Q&A Getting Started] [answered]
### Check these boxes if you have followed the posting rules.

- [X] Q&A Getting Started is the most appropriate category for my question (trouble installing, beginner user, ...)
- [X] I have consulted the posting Guidelines on the Discussions front page
- [X] I have searched the Discussions forum and my question has not been asked before
- [X] I have searched the MOOSE website and the documentation does not answer my question
- [X] I have formatted my post following the posting guidelines (avoid screenshots if possible, triple back quotes before/after pasted text, etc)

### Issue or question about MOOSE

I want to install MOOSE on my Mac (M2) for the first time. I faced this error. How can I solve this problem?
Channels:
 - https://conda.software.inl.gov/public
 - defaults
Platform: osx-arm64
Collecting package metadata (repodata.json): done
Solving environment: failed

LibMambaUnsatisfiableError: Encountered problems while solving:
  - nothing provides livereload needed by moose-tools-2024.05.02-hd1db04d_0

Could not solve for environment specs
The following package could not be installed
└─ moose-dev 2024.05.13**  is not installable because it requires
   └─ moose-tools 2024.05.02.* , which requires
      └─ livereload, which does not exist (perhaps a missing channel).

### Diagnostics for MOOSE installation issues

[Optional] Output of the diagnostics scripts. To run the script: cd ~/projects/moose/scripts; ./diagnostics.sh

**Comment by GiudGiud** [comment]
@milljm @loganharbour

**Reply by milljm** [reply, marked as answer]
This may be due to using a version of Conda that which does not by default support the `conda-forge` channel (Anaconda's 'Miniconda' implementation for example).

Try running the following `conda` commands to add the appropriate channels, and the necessary attributes to make use of these channels:
```bash
conda config --add channels conda-forge
conda config --add channels https://conda.software.inl.gov/public  # you'll probably get a warning about moving this to the top.
conda config --set channel_priority strict
``` 
And then try to install `moose-dev` once more.

**Reply by lijiaoyan** [reply]
I encountered the same issue and the issue was solved by following these steps. Thank you.

**Comment by DavoodZivar** [comment]
Thanks for your reply. It worked. When I wanted to run an example, I faced: 
```
(base) davoodzivar@Davoods-MacBook-Air ~ % cd ~/projects/moose/examples/ex01_inputfile
make -j8
./ex01-opt -i ex01.i
/bin/sh: /Users/davoodzivar/projects/moose/libmesh/installed/contrib/bin/libmesh-config: No such file or directory
/bin/sh: /Users/davoodzivar/projects/moose/libmesh/installed/contrib/bin/libmesh-config: No such file or directory
/bin/sh: /Users/davoodzivar/projects/moose/libmesh/installed/contrib/bin/libmesh-config: No such file or directory
/bin/sh: /Users/davoodzivar/projects/moose/libmesh/installed/contrib/bin/libmesh-config: No such file or directory
/bin/sh: /Users/davoodzivar/projects/moose/libmesh/installed/contrib/bin/libmesh-config: No such file or directory
/bin/sh: /Users/davoodzivar/projects/moose/libmesh/installed/contrib/bin/libmesh-config: No such file or directory
/bin/sh: /Users/davoodzivar/projects/moose/libmesh/installed/contrib/bin/libmesh-config: No such file or directory
/bin/sh: /Users/davoodzivar/projects/moose/libmesh/installed/contrib/bin/libmesh-config: No such file or directory
/bin/sh: /Users/davoodzivar/projects/moose/libmesh/installed/contrib/bin/libmesh-config: No such file or directory
/bin/sh: /Users/davoodzivar/projects/moose/libmesh/installed/libtool: No such file or directory
/bin/sh: /Users/davoodzivar/projects/moose/libmesh/installed/libtool: No such file or directory
/bin/sh: /Users/davoodzivar/projects/moose/libmesh/installed/contrib/bin/libmesh-config: No such file or directory
/bin/sh: /Users/davoodzivar/projects/moose/libmesh/installed/contrib/bin/libmesh-config: No such file or directory
/bin/sh: /Users/davoodzivar/projects/moose/libmesh/installed/contrib/bin/libmesh-config: No such file or directory
find: /Users/davoodzivar/projects/moose/framework/contrib/wasp/install/lib: No such file or directory
/bin/sh: /Users/davoodzivar/projects/moose/libmesh/installed/contrib/bin/libmesh-config: No such file or directory
Linking Test Library /Users/davoodzivar/projects/moose/examples/ex01_inputfile/test/lib/libex01_test-opt.la...
make: /Users/davoodzivar/projects/moose/libmesh/installed/libtool: No such file or directory
make: *** [/Users/davoodzivar/projects/moose/examples/ex01_inputfile/test/lib/libex01_test-opt.la] Error 1
make: *** Waiting for unfinished jobs....
zsh: no such file or directory: ./ex01-opt
  ```
How can I solve it?

**Reply by GiudGiud** [reply]
Hello

If you have built libmesh then you should export the LIBMESH_DIR environment variable

Use the folder where you built libmesh

Guillaume

**Comment by DavoodZivar** [comment]
I can run examples from MOOSE directory but when I try to run my code it says:  How can I solve this problem?

*** ERROR ***
A 'SmoothCircleIC' is not a registered object.

If you are trying to find this object in a dynamically loaded library, make sure that
the library can be found either in your "Problem/library_path" parameter or in the
MOOSE_LIBRARY_PATH environment variable.

Stack frames: 20
0: 0   libmesh_opt.0.dylib                 0x0000000106de7158 libMesh::print_trace(std::__1::basic_ostream<char, std::__1::char_traits<char>>&) + 1044
1: 1   libmoose-opt.0.dylib                0x0000000105d991c8 moose::internal::mooseErrorRaw(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>) + 680
2: 2   libmoose-opt.0.dylib                0x0000000104e8c5a0 void mooseError<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>>(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>&&) + 224
3: 3   libmoose-opt.0.dylib                0x0000000105d81440 Factory::reportUnregisteredError(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&) const + 416
4: 4   libmoose-opt.0.dylib                0x0000000105d817a8 Factory::getValidParams(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&) const + 432
5: 5   libmoose-opt.0.dylib                0x0000000105a315bc MooseObjectAction::MooseObjectAction(InputParameters const&) + 276
6: 6   libmoose-opt.0.dylib                0x0000000105a71fe0 RegistryEntry<AddICAction>::buildAction(InputParameters const&) + 64
7: 7   libmoose-opt.0.dylib                0x0000000105a307bc ActionFactory::create(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&, InputParameters&) + 976
8: 8   libmoose-opt.0.dylib                0x0000000105818230 Moose::Builder::walkRaw(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, hit::Node*) + 1256
9: 9   libmoose-opt.0.dylib                0x0000000105880c2c Moose::Builder::walk(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&, hit::Node*) + 268
10: 10  libhit-opt.0.dylib                  0x000000010301fca8 hit::Node::walk(hit::Walker*, hit::NodeType, hit::TraversalOrder) + 216
11: 11  libhit-opt.0.dylib                  0x000000010301fce4 hit::Node::walk(hit::Walker*, hit::NodeType, hit::TraversalOrder) + 276
12: 12  libhit-opt.0.dylib                  0x000000010301fce4 hit::Node::walk(hit::Walker*, hit::NodeType, hit::TraversalOrder) + 276
13: 13  libhit-opt.0.dylib                  0x000000010301fce4 hit::Node::walk(hit::Walker*, hit::NodeType, hit::TraversalOrder) + 276
14: 14  libmoose-opt.0.dylib                0x0000000105884a50 Moose::Builder::build() + 2336
15: 15  libmoose-opt.0.dylib                0x0000000105dac13c MooseApp::setupOptions() + 9768
16: 16  libmoose-opt.0.dylib                0x0000000105da96b8 MooseApp::run() + 1524
17: 17  ex01-opt                            0x0000000102acbbec void Moose::main<ExampleApp>(int, char**) + 156
18: 18  ex01-opt                            0x0000000102acbb44 main + 12
19: 19  dyld                                0x00000001868be0e0 start + 2360
application called MPI_Abort(MPI_COMM_WORLD, 1) - process 0
[unset]: write_line error; fd=-1 buf=:cmd=abort exitcode=1
:
system msg for write_line failure : Bad file descriptor

**Comment by GiudGiud** [comment]
Hello

You need to recompile with the phase field module I think
You add this in the makefile

**Reply by DavoodZivar** [reply]
As I am new to this environment, can you explain how I can do this?

**Reply by GiudGiud** [reply]
The Makefile should be in the root folder of your application.
Once you open it with a text editor, you can see the list of modules compiled with your application. 

SmoothCircleIC is in the phase_field module, so add 'YES' in front of that module
```
PHASE_FIELD                 := yes
```

then make clobberall then recompile

**Reply by DavoodZivar** [reply]
Thanks for your help.
Makefile shows this item was sat to "yes" but still the same error is there. Is there any other way to solve this?

**Reply by GiudGiud** [reply]
Unfortunately you do need to clean the repo and recompile between having it set to 'no' and set to 'yes'

so 
```
cd ~/projects/<name of the app>
git commit -am "saving everything"
git clean -xfD
make -j 6
```

**Reply by DavoodZivar** [reply]
Thanks for your help. I sat it to 'no' and then ran the comments you provided, then I sat it again to 'yes' and again ran the provided comments. After that, still the error is there. Was what I did correct?

**Reply by GiudGiud** [reply]
Weird. 
You could try cleaning moose too, not that it should be needed.
```
cd ~/projects/moose
git clean -xfd
cd ~/project/your_app
make -j 6
```

Else a fresh clone could help.

**Reply by DavoodZivar** [reply]
After several trials, I could add phase field module. Now, I need to add solid mechanics as well but I was not enable to it. I tried previous comments that you suggested but it did not work. No phase filed module works but solid mechanics no!
Can you help me with this?

**Reply by GiudGiud** [reply]
hello

the procedure is the same for both modules. 
once added in the makefile, you commit that, then clean everything, then re-build
